types/hardwarearchitecture.pp

```diff
@@ -2,4 +2,4 @@
 # @summary
 #   This custom type is used to manage the list of architecture of a provisioned node.
 #
-type Provision::Architecture = Enum['arm', 'amd', 'intel']
+type Provision::HardwareArchitecture = Enum['arm', 'amd', 'intel']
```

Comment:
can the filename have an underscore like `hardware_architecture.pp`?

Reply:
that way it doesn't work, this is default convention for any custom types.
FYI : https://www.puppet.com/docs/puppet/5.5/lang_type_aliases.html